### PR TITLE
add extension_list example

### PR DESCRIPTION
**Pull request by yangjie-msft**
created an example of agent extension policy that is used for Azure agent extensions.
added support for .NET framework 4.0 binding.
added a pytest test for the policy using the python bindings (thus also tested python binding)
Reorganized the structure accordingly due to the above changes.

### CODE REVIEW

File: bindings/csharp/net40/Program.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿using System;
```
Comment (mgunnala) [inline bindings/csharp/net40/Program.cs:1]: should we add a proper C# test suite before merging to main?

Comment (yangjie-msft) [reply bindings/csharp/net40/Program.cs:1]: we could do it in later stages. there are a couple of things we can do but let's first have the base line into the main.

**Review by mgunnala — APPROVED** [review]

File: .github/workflows/test-csharp.yml [hunk 2]
```diff
@@ -19,8 +19,8 @@ jobs:
 
       - name: Build
         run: dotnet build
-        working-directory: ./bindings/csharp
+        working-directory: ./bindings/csharp/net8.0
```
Comment (anakrish) [inline .github/workflows/test-csharp.yml:22]: We could test both net8.0 and net40

File: bindings/csharp/net40/Program.cs (hunk 1 above)
Comment (anakrish) [inline bindings/csharp/net40/Program.cs:1]: Add copyright notice in each file

File: examples/extension_list/agent_extension_policy.rego [hunk 3]
```diff
@@ -0,0 +1,122 @@
+package agent_extension_policy
```
Comment (anakrish) [inline examples/extension_list/agent_extension_policy.rego:1]: Add copyright notice for each source file.

**Review by anakrish — COMMENTED** [review]
Looks good for most part.

It looks like net40 is not being used. We should either lock it down with tests of remove it altogether if it is not needed.

**Comment by yangjie-msft** [comment]
> Looks good for most part.
> 
> It looks like net40 is not being used. We should either lock it down with tests of remove it altogether if it is not needed.

net40 is not tested because I am not sure if the tool chain can support net8.0 and net40 at the same time as net40 is pretty old but still needed by some applications.

File: .github/workflows/test-csharp.yml (hunk 2 above)
Comment (yangjie-msft) [reply .github/workflows/test-csharp.yml:22]: I can try, but not sure if the tool chain can support both of them at the same time

File: examples/extension_list/agent_extension_policy.rego (hunk 3 above)
Comment (yangjie-msft) [reply examples/extension_list/agent_extension_policy.rego:1]: will do.

File: bindings/csharp/net40/Program.cs (hunk 1 above)
Comment (yangjie-msft) [reply bindings/csharp/net40/Program.cs:1]: will do.

**Pull request closed without merging**